src/codegen/mod.rs

```diff
@@ -2105,8 +2109,26 @@ impl CodeGenerator for CompInfo {
 
                                         Some(quote! {
                                             assert_eq!(
-                                                unsafe {
-                                                    &(*(::#prefix::ptr::null::<#canonical_ident>())).#field_name as *const _ as usize
+                                                {
+                                                    // Create an instance of #canonical_ident struct from zero bit pattern
+                                                    const STRUCT_SIZE: usize = std::mem::size_of::<#canonical_ident>();
+                                                    let buffer = [0u8; STRUCT_SIZE];
```

Comment:
This is not guaranteed to be aligned right? I think using std::mem::zeroed should be better.

Reply:
Thanks, fixed

Though, I don't catch why there is a wrong alignment. AFAIU, In case of mem::zeroed the padding bytes are not zeroed but the total behavior is the same.

Reply:
No, let's say you have a struct with an `u32` member. The compiler would make sure that that is aligned appropriately, but if you use a flat array of `u8`s then it's not.

Reply:
@emilio 
Ok, thank you. Replaced it with mem::zeroed